compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -1037,15 +1037,14 @@ macro_rules! visit_place_fns {
 
         fn super_projection(
             &mut self,
-            local: Local,
-            projection: &[PlaceElem<'tcx>],
+            place_ref: PlaceRef<'tcx>,
             context: PlaceContext,
             location: Location,
         ) {
-            let mut cursor = projection;
+            let mut cursor = place_ref.projection;
             while let &[ref proj_base @ .., elem] = cursor {
```

Comment:
Can't this use `Place::iter_projections` (reversed)? Oh, that's on a place, not a `PlaceRef`... rather silly indeed.
Let's leave that for a future PR, but could you add a FIXME here, like
```
// FIXME: Use PlaceRef::iter_projections, once that exists.
```

Reply:
(At that point, `visit_projection_elem` should probably also be changed to use `PlaceRef`. But let's first land the work you did so far. :)